code/datums/ai_laws.dm

```diff
@@ -24,170 +56,338 @@
 			return ai_law
 	return null
 
+
+// ********* Getters for law descriptions *********
+
+/datum/ai_laws/proc/get_desc_devil_law_sixsixsix(index)
+	var/ret = null
+	if (index < 1 || index > devil_laws.len)
+		return ret
+	if (devil_laws[index])
+		var/datum/ai_law/law = devil_laws[index]
+		ret = law.description
+	return ret
+
+/datum/ai_laws/proc/get_desc_supplied(index)
+	var/ret = null
+	if (index < 1 || index > supplied.len)
+		return ret
+	if (supplied[index])
+		var/datum/ai_law/law = supplied[index]
+		ret = law.description
+	return ret
+
+/datum/ai_laws/proc/get_desc_inherent(index)
+	var/ret = null
+	if (index < 1 || index > inherent.len)
+		return ret
+	if (inherent[index])
+		var/datum/ai_law/law = inherent[index]
+		ret = law.description
+	return ret
+
+/datum/ai_laws/proc/get_desc_ion(index)
+	var/ret = null
+	if (index < 1 || index > ion.len)
+		return ret
+	if (ion[index])
+		var/datum/ai_law/law = ion[index]
+		ret = law.description
+	return ret
+
+/datum/ai_laws/proc/get_desc_zeroth()
+	var/ret = null
+	if (zeroth)
+		ret = zeroth.description
+	return ret
+
+/datum/ai_laws/proc/get_desc_zeroth_borg()
+	var/ret = null
+	if (zeroth_borg)
+		ret = zeroth_borg.description
+	return ret
+
+// ********* Getters and Setters for law states *********
+
+/datum/ai_laws/proc/get_state_devil_law_sixsixsix(index)
+	var/ret = null
+	if (index < 1 || index > devil_laws.len)
+		return ret
+	if (devil_laws[index])
+		var/datum/ai_law/law = devil_laws[index]
+		ret = law.state
+	return ret
+
+/datum/ai_laws/proc/get_state_supplied(index)
+	var/ret = null
+	if (index < 1 || index > supplied.len)
+		return null
+	if (supplied[index])
+		var/datum/ai_law/law = supplied[index]
+		ret = law.state
+	return ret
+
+/datum/ai_laws/proc/get_state_inherent(index)
+	var/ret = null
+	if (index < 1 || index > inherent.len)
+		return ret
+	if (inherent[index])
+		var/datum/ai_law/law = inherent[index]
+		ret = law.state
+	return ret
+
+/datum/ai_laws/proc/get_state_ion(index)
+	var/ret = null
+	if (index < 1 || index > ion.len)
+		return ret
+	if (ion[index])
+		var/datum/ai_law/law = ion[index]
+		ret = law.state
+	return ret
+
+/datum/ai_laws/proc/get_state_zeroth()
+	var/ret = null
+	if (zeroth)
+		ret = zeroth.state
+	return ret
+
+/datum/ai_laws/proc/get_state_zeroth_borg()
+	var/ret = null
+	if (zeroth_borg)
+		ret = zeroth_borg.state
+	return ret
+
+
+// The setters return 1 if there's a problem, returns 0 if no problem
+
+/datum/ai_laws/proc/set_state_devil_law_sixsixsix(index, new_state)

```

Comment:
Keep in mind this would mean it's returning TRUE in the event that it fails and FALSE if it succeeds.